Sources/Playbook/SnapshotSupport/SnapshotSupport.swift

```diff
@@ -71,71 +71,130 @@ public enum SnapshotSupport {
+    ///   - keyWindow: The key window of the application.
+    ///   - handler: A closure that to handle generated `UIView`.
+    ///
+    /// - Note: Passing the key window adds the scenario content to the view
+    ///         hierarchy and actually renders it, so producing a more accurate
+    ///         snapshot image.
+    public static func view(
         for scenario: Scenario,
         on device: SnapshotDevice,
-        scale: CGFloat,
-        keyWindow: UIWindow?,
-        completion: @escaping (Resource) -> Void
+        keyWindow: UIWindow? = nil,
+        handler: @escaping (UIView) -> Void
     ) {
         withoutAnimation {
             let window = SnapshotWindow(scenario: scenario, device: device)
             let contentView = window.contentView!
 
-            let isEmbedInKeyWindow: Bool
-
             if let keyWindow = keyWindow {
                 keyWindow.addSubview(window)
-                isEmbedInKeyWindow = true
-            }
-            else {
-                isEmbedInKeyWindow = false
             }
 
             window.prepareForSnapshot {
                 if contentView.bounds.size.width <= 0 {
-                    fatalError("The view did laid out with zero width in scenario - \(scenario.name)", file: scenario.file, line: scenario.line)
+                    fatalError("The view was laid out with zero width in scenario - \(scenario.name)", file: scenario.file, line: scenario.line)
                 }
 
                 if contentView.bounds.size.height <= 0 {
-                    fatalError("The view did laied out with zero height in scenario - \(scenario.name)", file: scenario.file, line: scenario.line)
+                    fatalError("The view was laid out with zero height in scenario - \(scenario.name)", file: scenario.file, line: scenario.line)
                 }
 
-                let format = UIGraphicsImageRendererFormat(for: device.traitCollection)
-                format.scale = scale
+                handler(contentView)
+            }
+        }
+    }
 
-                if #available(iOS 12.0, *) {
-                    format.preferredRange = .standard
+    /// Generates an image file data for the given `UIView`
+    ///
+    /// - Parameters:
+    ///   - view: A `UIView` of a snapshot.
+    ///   - device: A snapshot environment simulating device.
+    ///   - format: An image file format of exported data.
+    ///   - scale: A rendering scale of the snapshot image.
+    ///   - keyWindow: The key window of the application.
+    ///   - handler: A closure that to handle generated data.
+    ///
+    /// - Note: Passing the key window adds the scenario content to the view
+    ///         hierarchy and actually renders it, so producing a more accurate
+    ///         snapshot image.
+    public static func data(
+        for view: UIView,
+        on device: SnapshotDevice,
+        format: ImageFormat,
+        scale: CGFloat = UIScreen.main.scale,
+        keyWindow: UIWindow? = nil,
```

Comment:
maybe we can go an API like the folowing instead of `static func view(for:...)`:

```diff
public static func data(
    for view: UIView,
    on device: SnapshotDevice,
    format: ImageFormat,
    scale: CGFloat = UIScreen.main.scale,
    keyWindow: UIWindow? = nil,
+    viewPreprocessor: ((UIView) -> UIView)? = nil,
    handler: @escaping (Data) -> Void
) {}
```